common-lib/src/main/scala/com/gu/mediaservice/lib/json/JsonByteArrayUtil.scala

```diff
@@ -0,0 +1,55 @@
+package com.gu.mediaservice.lib.json
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import java.nio.charset.StandardCharsets
+import java.util.zip.{GZIPInputStream, GZIPOutputStream}
+
+import play.api.libs.json._
+
+import scala.io.Source.fromInputStream
+
+object JsonByteArrayUtil extends PlayJsonHelpers {
+  private val compressionMarkerByte: Byte = 0x00.toByte
+
+  private def compress(bytes: Array[Byte]): Array[Byte] = {
+    val outputStream = new ByteArrayOutputStream()
+    val zipOutputStream = new GZIPOutputStream(outputStream)
+    zipOutputStream.write(bytes)
+    zipOutputStream.close()
+    outputStream.close()
+    val compressedBytes = outputStream.toByteArray
+    compressionMarkerByte +: compressedBytes
+  }
+
+  private def decompress(bytes: Array[Byte]): Array[Byte] = {
+    val bytesWithoutCompressionMarker = bytes.tail
+    val byteStream = new ByteArrayInputStream(bytesWithoutCompressionMarker)
+    val inputStream = new GZIPInputStream(byteStream)
+    val decompressedBytes = fromInputStream(inputStream).mkString.getBytes
+    byteStream.close()
+    inputStream.close()
+    decompressedBytes
+  }
+
+  def hasCompressionMarker(bytes: Array[Byte]) = bytes.head == compressionMarkerByte
+
+  def toByteArray[T](obj: T, withCompression: Boolean = false)(implicit writes: Writes[T]): Array[Byte] = {
+    val bytes = Json.toBytes(Json.toJson(obj))
+    if (withCompression) compress(bytes) else bytes
+  }
+
+  def fromByteArray[T](bytes: Array[Byte])(implicit reads: Reads[T]): Option[T] = {
+    val string = new String(
+      if (hasCompressionMarker(bytes)) decompress(bytes) else bytes,
```

Comment:
👍

Reply:
Worth noting this gets simplified in the other PRs https://github.com/guardian/grid/pull/2645 and https://github.com/guardian/grid/pull/2646.